lib_eio/unix/eio_unix.mli

```diff
@@ -107,6 +107,9 @@ module Private : sig
 
   val read_link : Fd.t option -> string -> string
   val read_link_unix : Unix.file_descr option -> string -> string
+
+  val chown : flags:int -> uid:int64 -> gid:int64 -> Fd.t -> string -> unit
+  val chown_unix : flags:int -> uid:int64 -> gid:int64 -> Unix.file_descr -> string -> unit
```

Comment:
I guess this is just copying the pattern above, but I think we only have `read_link_unix` because it's needed during path resolution and wrapping each step in `Fd.t` seemed unreasonable.

Reply:
Most of the low-level function in the posix backend use `Unix.file_descr` and this is in keeping with that? 

```ocaml
  Resolve.with_parent "<operation>" dirfd path @@ fun dirfd path ->
  let dirfd = Option.value dirfd ~default:at_fdcwd in
  ```
  
  They follow this pattern where `dirfd` ends up being a `Unix.file_descr` -- should we wrap them with `Fd.of_unix` + a switch?